### PR TITLE
[enhancement] make service type product bundle #3452

### PR DESCRIPTION
**Pull request by rmehta**

### CODE REVIEW

**Comment by inventobd** [comment]
Thanks man . you are awesome

File: erpnext/selling/doctype/product_bundle/product_bundle.py [hunk 1]
```diff
@@ -13,17 +13,9 @@ def autoname(self):
 		self.name = self.new_item_code
 
 	def validate(self):
-		self.validate_main_item()
-
 		from erpnext.utilities.transaction_base import validate_uom_is_integer
 		validate_uom_is_integer(self, "uom", "qty")
 
-	def validate_main_item(self):
```
Comment (nabinhait) [inline erpnext/selling/doctype/product_bundle/product_bundle.py]: Parent item must be non-stock item

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 2]
```diff
@@ -39,9 +39,8 @@ def validate_for_items(self):
 		for d in self.get('items'):
 			check_list.append(cstr(d.item_code))
 
-			if (frappe.db.get_value("Item", d.item_code, "is_stock_item")==1 or
-				self.has_product_bundle(d.item_code)) and not d.warehouse:
-					frappe.throw(_("Reserved warehouse required for stock item {0}").format(d.item_code))
+			if frappe.db.get_value("Item", d.item_code, "is_stock_item") and not d.warehouse:
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order/sales_order.py]: If child item of a Product Bundle is stock-item, then make reserve warehouse mandatory.

File: erpnext/stock/get_item_details.py [hunk 3]
```diff
@@ -224,6 +226,22 @@ def get_price_list_rate(args, item_doc, out):
...
+		and cint(frappe.db.get_single_value("Stock Settings", "auto_insert_price_list_rate_if_missing")):
+		if frappe.has_permission("Item Price", "write"):
+			item_price = frappe.get_doc({
+				"doctype": "Item Price",
+				"price_list": args.price_list,
+				"item_code": args.item_code,
+				"currency": args.currency,
+				"price_list_rate": args.rate
+			})
+			item_price.insert()
```
Comment (nabinhait) [inline erpnext/stock/get_item_details.py]: If item_price record already exists, then don't insert

Comment (rmehta) [inline erpnext/stock/get_item_details.py]: will only be called if price_list_rate is not defined

**Comment by rmehta** [comment]
fixed

**Pull request merged**